torch/fx/passes/dialect/common/cse_pass.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict, Tuple, Any
+
+import torch
+import torch.fx as fx
+from torch.fx.passes.infra.pass_base import PassBase, PassResult
+from torch.utils._pytree import tree_flatten
+
+aten = torch.ops.aten
+
+
+def get_aten_target(node):
+    if hasattr(node.target, 'overloadpacket'):
+        return node.target.overloadpacket
+    return node.target
+
+# random ops are banned from CSE
```

Comment:
it's not just rand ops, any stateful ops are actually not supported. 

consider following case: 

def fn(x):
    a = torch.aten.add_(x, 1) 
    b = torch.aten.add_(x, 1)
    return x

Reply:
We need to be super clear about the applicable condition for this pass. 

Please update the document for this.

Reply:
hmmm is there a better way to detect all of them without listing all manually?

Reply:
No better way to do it at this moment.... 
We want to say this CSE pass can only be applied on certain dialect... where the ops in this dialect are functional (non-stateful)... 

You will be able to do this when we have define the "Canonical/Functional Aten Opset".